### ISSUE TITLE
Order of font elements intermittently changes in CI

### ISSUE THREAD

**Issue by mikevoets** [mattbrictson/minitest-snapshots#55]
We've noticed that after upgrading to 1.1.0, our tests in CI intermittently flake due to the snapshot not being expected.

By further inspection it looks like inner elements of `font` elements sometimes change their order. See:

```
+ <font><u val=\"single\"/><name val=\"Arial\"/><sz val=\"11\"/><family val=\"1\"/></font>
- <font><sz val=\"11\"/><family val=\"1\"/><name val=\"Arial\"/><u val=\"single\"/></font>
```

Expected is that they'll keep the same order every time so the snapshots will match.

Is there anything on our end we can do to resolve this issue?

**Comment by mattbrictson** [comment]
Hmm, my first impression is that this is non-deterministic behavior in whatever code is generating the value that is being snapshotted. However, there is some "magic" in minitest-snapshots that attempts to sort the contents of Hash and Set elements before serializing them. It's not immediately clear to me if/how this could be causing a problem, but if you want to eliminate that as a possibility, you could serialize objects yourself before invoking the snapshot mechanism.

In other words, let's say you have a `document` that is an instance of `Nokogiri::XML`.

```ruby
# Implicit serialization
assert_matches_snapshot(document)

# Explicit serialization
assert_matches_snapshot(document.to_xml) # Convert to string before comparing
```

If you have a deterministic way to convert the object to a string (`.to_xml` in the above example), then that makes it much easier to guarantee that the snapshots will match.
